Move iOS view management to compose.ui

elijah-semyonov opened the pull request:
## Proposed Changes

Copy SkikoUIView from skiko, use local version instead.

## Testing

Test: TBD

## Depends on
https://github.com/JetBrains/skiko/pull/757

elijah-semyonov committed:
Copy SkikoUIView from skiko. Use it instead of skiko-exposed one.

MatkovIvan commented on compose/ui/ui/src/uikitMain/kotlin/org/jetbrains/skiko/bridge/SkikoUIView.kt:
```diff
@@ -0,0 +1,566 @@
+package org.jetbrains.skiko.bridge
```
Shouldn't it be placed into separate module for window management? or be a part of androidx.*

elijah-semyonov replied:
Discussed in internal slack channel, see Dima's recent question.

dima-avdeev-jb replied:
move to `package androidx.compose.ui.platform`

MatkovIvan reviewed (changes requested):
Blocking it for now. By splitting skiko we meant creating a separate module for window management

elijah-semyonov committed:
Add @OptIn(InternalSkikoApi::class) for SkikoUIView

dima-avdeev-jb commented on compose/ui/ui/src/uikitMain/kotlin/org/jetbrains/skiko/bridge/SkikoUIView.kt:
```diff
@@ -0,0 +1,567 @@
+@Suppress("CONFLICTING_OVERLOADS")
+@ExportObjCClass
+@OptIn(InternalSkikoApi::class)
+class SkikoUIView : UIView, UIKeyInputProtocol, UITextInputProtocol {
```
```suggestion
internal class SkikoUIView : UIView, UIKeyInputProtocol, UITextInputProtocol {
```

dima-avdeev-jb committed:
Add MetalContextHandler and MetalRedrawer (#633)

* Add MetalContextHandler and MetalRedrawer

* move abstract ContextHandler

* inline abstract ContextHandler in MetalContextHandler

* Merge MetalContextHandler.kt into MetalRedrawer.ios.kt

* Refactor MetalRedrawer class and move some responsibilities to SkikoUIView

* Use SkiaLayerInterface instead of SkiaLayer

elijah-semyonov committed:
Refactor MetalRedrawer and SkiaLayer2, remove redundant calls and dead code.

elijah-semyonov committed:
Poll drawable size from metalLayer.

elijah-semyonov committed:
Refactor MetalRedrawer and SkiaLayer2 classes.

This commit refactors the MetalRedrawer and SkiaLayer2 classes of the Skia library. The change includes making the context and queue properties final, the removal of the initContext() method and reordering of methods and properties in the MetalRedrawer class.

The refactoring is done to make code simpler and safer, by reducing mutable state and ensuring context is initialized correctly at object creation. The pixelGeometry property was removed from SkiaLayer2 since it was unused.

elijah-semyonov committed:
Refactor MetalRedrawer to encapsulate frame rendering info

Code refactored in MetalRedrawer class of Skiko library to create a new data class named 'FrameRenderBackendInfo' to bundle renderTarget, surface, and metalDrawable for a frame. This change simplifies the management of objects needed in frame rendering and eliminates unnecessary multiple variable declarations..

dima-avdeev-jb committed:
Merge commits from jb-main (#638)

* Add sample, How to hide Keyboard on click outside TextField (#631)

* Commonize Scrollbar in skikoMain from desktopMain (#571)

* Commonize Scrollbar in skikoMain from desktopMain

* Rename runBlockingCoroutine to runBlockingIfPossible

* Add `@file:JvmName("Scrollbar_desktopKt")`

* Add JvmName annotation related to Scrollbar commonization (#635)

@file:JvmName("Scrollbar_desktopKt")
@file:JvmName("TextFieldScrollState_desktop_ktKt")

* Implemented SkikoUITextInputTraits + example with textfields (#636)

* iOS TextField implement UIKitKeyboardOptions

* iOS TextField, implement SkikoUITextInputTraits

* iOS TextField, implement SkikoUITextInputTraits

* Added example with Keyboard actions

* Added example with password textfields

---------

Co-authored-by: dima.avdeev <dima.avdeev@jetbrains.com>

* use _skikoUITextInputTraits

---------

Co-authored-by: Oleksandr Karpovich <a.n.karpovich@gmail.com>
Co-authored-by: Vladimir Mazunin <124871594+mazunin-v-jb@users.noreply.github.com>

elijah-semyonov committed:
Refactor MetalRedrawer

elijah-semyonov committed:
Reorder members

elijah-semyonov committed:
Reorder control flow

elijah-semyonov committed:
Decouple MetalRedrawer from SkiaLayer

elijah-semyonov committed:
Decouple MetalRedrawer from SkikoLayer

dima-avdeev-jb committed:
Merge branch 'jb-main' into elijah.semyonov/move-skiko-to-core

# Conflicts:
#	compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt

elijah-semyonov committed:
Merge remote-tracking branch 'origin/elijah.semyonov/move-skiko-to-core' into elijah.semyonov/move-skiko-to-core

elijah-semyonov committed:
Reorder members

elijah-semyonov committed:
Explicitly pass skikoInput to skikoUIView

elijah-semyonov committed:
Rename SkiaLayer2 to IOSSkiaLayer

elijah-semyonov committed:
Revert "Explicitly pass skikoInput to skikoUIView"

This reverts commit 2c0825aa81da109dd943e8dda354322c7ad12451.

elijah-semyonov committed:
Delete dead SkikoGesturesDetector.kt code

elijah-semyonov committed:
Simplify SkikoUIView interaction with IOSSkiaLayer

elijah-semyonov committed:
Fix logical bug, make scene size update be driven by UIKit layout

elijah-semyonov committed:
Remove dbg println

elijah-semyonov committed:
Simplify code, add comments

elijah-semyonov committed:
Rename layer to composeLayer

elijah-semyonov committed:
Rearrange expression

elijah-semyonov committed:
Remove redundant code

elijah-semyonov committed:
Add remark

elijah-semyonov committed:
Update compose/ui/ui/src/uikitMain/kotlin/org/jetbrains/skiko/bridge/SkikoUIView.kt

Co-authored-by: dima.avdeev <99798741+dima-avdeev-jb@users.noreply.github.com>

elijah-semyonov committed:
Refactor MetalRedrawer, now using the device from metalLayer and adjusting the drawInCanvas function call. Removed the draw method inside the IOSSkiaLayer class, pushing the skikoView rendering directly inside SkikoUIView initialization. Also added a TODO comment about the IOSSkiaLayer not being disposed.

elijah-semyonov committed:
Merge remote-tracking branch 'origin/elijah.semyonov/move-skiko-to-core' into elijah.semyonov/move-skiko-to-core

elijah-semyonov committed:
Merge branch 'jb-main' into elijah.semyonov/move-skiko-to-core

elijah-semyonov committed:
Remove redundant space conversion

elijah-semyonov committed:
Remove redundant space conversion

elijah-semyonov committed:
Remove redundant `getTopLeftOffset`

elijah-semyonov committed:
Merge branch 'elijah.semyonov/fix-redundant-touches-space' into elijah.semyonov/move-skiko-to-core

# Conflicts:
#	compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeWindow.uikit.kt
